### ISSUE TITLE
git.tags is empty

### ISSUE THREAD

**Issue by saikatganguly** [ktoso/maven-git-commit-id-plugin#320]
Hi,

I am trying with this plugin to get all available tags. My setting is as follows : 
```xml
		<configuration>
                   <dotGitDirectory>${project.basedir}/.git</dotGitDirectory>
                   <verbose>true</verbose>
                   <generateGitPropertiesFile>true</generateGitPropertiesFile>
                   <generateGitPropertiesFilename>${project.build.outputDirectory}/git.properties</generateGitPropertiesFilename>
                   <format>properties</format>
                   <skipPoms>false</skipPoms>
                   <injectAllReactorProjects>true</injectAllReactorProjects>
                   <failOnNoGitDirectory>true</failOnNoGitDirectory>
                   <failOnUnableToExtractRepoInfo>true</failOnUnableToExtractRepoInfo>
                   <skip>false</skip>
                   <runOnlyOnce>true</runOnlyOnce>
                   <useNativeGit>false</useNativeGit>
                   <gitDescribe>
                        <skip>false</skip>
                        <always>true</always>
                        <tags>true</tags>
                   </gitDescribe>                                  
                </configuration>
```

In my generated git.properties I can find git.closest.tag.name. But git .tags is empty. What I am trying to achieve is, I want to create build according to some tag. So I will pull all the tags and match them with given criteria and use in project version.

I checked in git cli with `git tag` command. That is showing all available tags.

Thanks!
Saikat

**Comment by TheSnoozer** [comment]
Hi thanks for reporting your problem here.
Couple of questions:
* What CI do you use and what version?
* are you using `lightweight tags` or `annotated tags` ?
* for the `tags` property we basically execute [git tag --contains](https://git-scm.com/docs/git-tag#git-tag---containsltcommitgt). Based on the documentation it would mean `Only list tags which contain the specified commit`. Are you sure your current `HEAD` or the tag you have checked out is on a tag? What is the output of `git describe --tags --long --dirty` ?

Can you elaborate on how your build is dependant on tags? This sounds like you are looking for profiles and what to control your build on profiles you execute and not on conditions on the underlying repository....

**Comment by saikatganguly** [comment]
Hi thanks for your reply. `git describe --tags --long --dirty` shows only the current tag which is exposed via git.closest.tag.name. I am using git command line. Git version is 1.9.5. I am using both lightweight tags and annotated tags

**Comment by TheSnoozer** [comment]
Hi,
from my perspective the plugin is working as expected.
Please checkout the documentation about what is supposed to be inside the `git.tags`-Property:
for the `git.tags` property we basically execute [git tag --contains](https://git-scm.com/docs/git-tag#git-tag---containsltcommitgt). Based on the documentation it would mean `Only list tags which contain the specified commit`. 
Since your commit is not included in a tag the property is empty (and thus this is expected behaviour).

To get a better understanding:
The `git.tags ` is pretty much the same feature what github has where it shows for each commit in what tags those are included.

![example](https://user-images.githubusercontent.com/6849390/30434691-e40e96e6-9935-11e7-9232-037fb8e4e1dc.png)

Obtained from here:
https://github.com/ktoso/maven-git-commit-id-plugin/commit/e808f91d596fd40ec7859dcf561bbbf13b676a66

## TLDR
Can you elaborate on how your build is dependant on tags or in general what you try to achive? 
This sounds like you are looking for profiles and want to control your build on profiles you execute and not on conditions on the underlying repository.

**Comment by TheSnoozer** [comment]
I'm closing this since its not a bug in the plugin
Feel free to post follow up questions on what you achieve

~Closed~
